### PR TITLE
Add documentation for SEM

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
This adds the docs for https://github.com/pingcap/tidb/blob/master/docs/design/2021-03-09-security-enhanced-mode.md

Which was merged in a series of PRs to master. The exact restrictions may evolve over time, so the wording is kept to use simple language to say "restrictions include". It is intended for use in cloud environments where the "root" privilege is considered too powerful.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/5647 https://github.com/pingcap/docs/pull/5747
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
 - 默认值：ON
 - 这个变量用于控制是否同时将各个执行算子的执行信息记录入 slow query log 中。
 
+### `tidb_enable_enhanced_security`
+
+- 作用域：NONE
+- 默认值：OFF
+- 这个变量表示所连接的 TiDB 服务器是否启用了安全增强模式（SEM），在不重新启动 TiDB 服务器的情况下不能改变该变量。
+- 安全增强模式的想法来自于诸如 [安全增强式 Linux](https://zh.wikipedia.org/wiki/%E5%AE%89%E5%85%A8%E5%A2%9E%E5%BC%BA%E5%BC%8FLinux) 等系统的设计。SEM 降低了具有 MySQL `SUPER` 权限的用户的能力，而是需要授予 `RESTRICTED` 细粒度权限作为替代。`RESTRICTED` 权限如下：
+    - `RESTRICTED_TABLES_ADMIN`：能够写入 `mysql` 模式中的系统表，并查看 `information_schema` 表上的敏感列。
```
Comment (SunRunAway) [inline system-variables.md:461]:
```suggestion
    - `RESTRICTED_TABLES_ADMIN`：能够写入 `mysql` 库中的系统表，并查看 `information_schema` 表上的敏感列。
```

File: system-variables.md [hunk 2]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
...
+    - `RESTRICTED_STATUS_ADMIN`：能够在 `SHOW STATUS` 命令中查看敏感变量。
```
Comment (SunRunAway) [inline system-variables.md:462]:
```suggestion
    - `RESTRICTED_STATUS_ADMIN`：能够在 `SHOW STATUS` 命令中查看敏感内容。
```

File: system-variables.md [hunk 3]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
...
+    - `RESTRICTED_VARIABLES_ADMIN`：能够在 `SHOW [GLOBAL] VARIABLES` 和 `SET` 命令中查看和设置敏感变量。
```
Comment (SunRunAway) [inline system-variables.md:463]:
```suggestion
    - `RESTRICTED_VARIABLES_ADMIN`：能够在 `SHOW [GLOBAL] VARIABLES` 和 `SET` 命令中查看和设置具有敏感内容的变量。
```

**Comment by TomShawn** [comment]
@SunRunAway 请问这个功能是要带入 v5.1 的吗？

**Comment by SunRunAway** [comment]
> @SunRunAway @bb7133 请问这个功能是要带入 v5.1 的吗？

要

**Comment by TomShawn** [comment]
@SunRunAway 能请再看看吗？谢谢~

File: system-variables.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ TiDB 系统变量的行为与 MySQL 相似但有一些不同，变量的作用
 
 - 对 `GLOBAL` 作用域变量的更改，设置后**只对新 TiDB 连接会话生效**，当前活动连接会话不受影响。更改会被持久化，重启后仍然生效。
 - 对 `INSTANCE` 作用域变量的更改，设置后会立即对当前 TiDB 实例所有活动连接会话或新连接会话生效，其他 TiDB 实例不生效。更改**不会**被持久化，重启 TiDB 后会**失效**。
+- 变量也可以有 `NONE` 作用域。这些变量是只读的，通常用于传递 TiDB 服务器启动后不会改变的静态信息。
```
Comment (SunRunAway) [inline system-variables.md:12]:
```suggestion
- 变量也可以有 `NONE` 作用域。这些变量是只读的，通常用于展示 TiDB 服务器启动后不会改变的静态信息。
```

**Review by SunRunAway — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -9,6 +9,7 @@ TiDB 系统变量的行为与 MySQL 相似但有一些不同，变量的作用
 
 - 对 `GLOBAL` 作用域变量的更改，设置后**只对新 TiDB 连接会话生效**，当前活动连接会话不受影响。更改会被持久化，重启后仍然生效。
 - 对 `INSTANCE` 作用域变量的更改，设置后会立即对当前 TiDB 实例所有活动连接会话或新连接会话生效，其他 TiDB 实例不生效。更改**不会**被持久化，重启 TiDB 后会**失效**。
+- 变量也可以有 `NONE` 作用域。这些变量是只读的，通常用于展示 TiDB 服务器启动后不会改变的静态信息。
```
Comment (TomShawn) [inline system-variables.md:12]:
```suggestion
- 作用域为 `NONE` 的变量为只读变量，通常用于展示 TiDB 服务器启动后不会改变的静态信息。
```

File: system-variables.md [hunk 6]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
 - 默认值：ON
 - 这个变量用于控制是否同时将各个执行算子的执行信息记录入 slow query log 中。
 
+### `tidb_enable_enhanced_security`
+
+- 作用域：NONE
+- 默认值：OFF
+- 这个变量表示所连接的 TiDB 服务器是否启用了安全增强模式（SEM），在不重新启动 TiDB 服务器的情况下不能改变该变量。
```
Comment (TomShawn) [inline system-variables.md:459]:
```suggestion
- 这个变量表示所连接的 TiDB 服务器是否启用了安全增强模式 (SEM)。若要改变该变量值，你需要在 TiDB 服务器的配置文件中修改 `enable-sem` 项的值，并重启 TiDB 服务器。
```

File: system-variables.md [hunk 7]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
...
+- 安全增强模式的想法来自于诸如 [安全增强式 Linux](https://zh.wikipedia.org/wiki/%E5%AE%89%E5%85%A8%E5%A2%9E%E5%BC%BA%E5%BC%8FLinux) 等系统的设计。SEM 降低了具有 MySQL `SUPER` 权限的用户的能力，而是需要授予 `RESTRICTED` 细粒度权限作为替代。`RESTRICTED` 权限如下：
```
Comment (TomShawn) [inline system-variables.md:460]:
```suggestion
- 安全增强模式受[安全增强式 Linux](https://zh.wikipedia.org/wiki/安全增强式Linux) 等系统设计的启发，削减拥有 MySQL `SUPER` 权限的用户能力，转而使用细粒度的 `RESTRICTED` 权限作为替代。这些细粒度的 `RESTRICTED` 权限如下：
```

File: system-variables.md [hunk 8]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
...
+    - `RESTRICTED_TABLES_ADMIN`：能够写入 `mysql` 库中的系统表，并查看 `information_schema` 表上的敏感列。
```
Comment (TomShawn) [inline system-variables.md:461]:
```suggestion
    - `RESTRICTED_TABLES_ADMIN`：能够写入 `mysql` 库中的系统表，能查看 `information_schema` 表上的敏感列。
```

File: system-variables.md [hunk 9]
```diff
@@ -451,6 +452,17 @@ mysql> SELECT * FROM t1;
...
+    - `RESTRICTED_STATUS_ADMIN`：能够在 `SHOW STATUS` 命令中查看敏感内容。
+    - `RESTRICTED_VARIABLES_ADMIN`：能够在 `SHOW [GLOBAL] VARIABLES` 和 `SET` 命令中查看和设置具有敏感内容的变量。
```
Comment (TomShawn) [inline system-variables.md:463]:
```suggestion
    - `RESTRICTED_VARIABLES_ADMIN`：能够在 `SHOW [GLOBAL] VARIABLES` 和 `SET` 命令中查看和设置包含敏感内容的变量。
```

File: tidb-configuration-file.md [hunk 10]
```diff
@@ -262,6 +262,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 
 安全相关配置。
 
+### `enable-sem`
+
+- 启用安全增强模式（SEM）。
```
Comment (TomShawn) [inline tidb-configuration-file.md:267]:
```suggestion
- 启用安全增强模式 (SEM)。
```

File: tidb-configuration-file.md [hunk 11]
```diff
@@ -262,6 +262,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
+- 默认值：`false`
+- 可以通过系统变量[`tidb_enable_enhanced_security`]（/system-variables.md#tidb_enable_enhanced_security）获得安全增强模式的状态。
```
Comment (TomShawn) [inline tidb-configuration-file.md:269]:
```suggestion
- 可以通过系统变量 [`tidb_enable_enhanced_security`](/system-variables.md#tidb_enable_enhanced_security) 获取安全增强模式的状态。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**